server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -1747,6 +1747,14 @@ public boolean restartVpc(final long vpcId, final boolean cleanUp, final boolean
                 return true;
             }
 
+            // Restart all networks in this VPC that needs a restart
+            List<? extends Network> networks = _ntwkModel.listNetworksByVpc(vpcId);
+            for (Network network: networks){
+                if (network.isRestartRequired()){
+                    _ntwkMgr.restartNetwork(network.getId(), callerAccount, callerUser, cleanUp);
+                }
+            }
```

Comment:
instead of a comment can you extract this into a method with good name, please?

Reply:
Done